languages/csharp/exercises/concept/regular-expressions/.docs/instructions.md

```diff
@@ -0,0 +1,100 @@
+This exercise addresses the parsing of log files.
+
+After a recent security review you have been asked to clean up the organisation's archived log files.
+
+All strings passed to the methods of guaranteed to be non-null and leading and trailing spaces are guaranteed to be removed.
+
+### 1. Identify garbled log lines
+
+You need some idea of how many log lines in your archive do not comply with current standards. You believe that a simple test reveals whether a log line is valid. To be considered valid a line should begin with one of the following strings:
+
+- [TRC]
+- [DBG]
+- [INF]
+- [WRN]
+- [ERR]
+- [FTL]
+
+Implement the `LogParser.IsValidLine()` method to return `false` if a string is not valid otherwise `true`.
+
+```csharp
+var lp = new LogParser();
+lp.IsValidLine("[ERR] A good error here");
+// => true
+lp.IsValidLine("Any old [ERR] text");
+// => false
+lp.IsValidLine("[BOB] Any old text");
+// => false
+```
+
+### 2. Split the log line
+
+A new team has joined the organization and you find their log files are using a strange separator for "fields". Instead of something sensible like a colon ":" they use a string such as "<--->" or "<=>" (because it's prettier) in fact any string that has a first character of "<" and a last character of ">" and any combination of the following "^\*=-" in between.
```

Comment:
I found the `"^\*=-"` bit somewhat hard to read. Maybe we could be a bit more verbose here and list the characters individually?

Reply:
Done